src/render/View3D.cpp

```diff
@@ -498,6 +514,42 @@ void View3D::paintGL()
     glCheckError();
 }
 
+
+QImage View3D::renderScreenShot(int width, int height)
+{
+    if (m_badOpenGL)
+        return QImage();
+
+    double dPR = getDevicePixelRatio();
+    if (width < 0)
+        width = this->width();
+
+    if (height < 0)
+        height = this->height();
+
+    // Note: the reason for the following two lines and the subsequent downsampling of the
+    //       resulting image is to make the radius of rendered points consistent between the
+    //       gui and screenshot on displas with dPR != 1.
+    width *= dPR;
+    height *= dPR;
+
+    QOpenGLFramebufferObject framebuffer(QSize(width, height), QOpenGLFramebufferObject::Depth);
+    framebuffer.bind();
+    resizeViewport(width, height);
```

Comment:
The reason for scaling up the offscreen buffer by dPR and subsequently downscaling the resulting image by the same factor was just a workaround to make points rendered into the offscreen buffer the same size as points in the gui when rendered on my Retina Macbook (dPR=2). If rendering to the offscreen buffer at the same resolution as the gui window, the points in the screenshot end up being twice as large as those on the screen. I made a short attempt at figuring out exactly why this was happening but had to get on with other things. Any idea why this would happen?